slamd/discovery/processing/experiment/models/tuned_gaussian_process_regressor.py

```diff
@@ -0,0 +1,63 @@
+from mlxtend.feature_selection import SequentialFeatureSelector
+from sklearn.gaussian_process import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import RBF, ConstantKernel
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import Pipeline
+
+
+class TunedGaussianProcessRegressor:
+
+    @classmethod
+    def find_best_model(cls, training_rows, training_labels):
+        # Perform a forward sequential feature selection with 3 features without cross-validation
+        sfs_gr_testing = SequentialFeatureSelector(estimator=cls._create_optimized_gaussian_process(),
+                                                   k_features=3,
+                                                   forward=True,
+                                                   floating=False,
+                                                   scoring='r2',
+                                                   cv=None)
+        pipe = Pipeline([('sfs', sfs_gr_testing),
+                         ('gp2', cls._create_optimized_gaussian_process())])
+        grid_search_cv = GridSearchCV(estimator=pipe,
+                                      param_grid=cls._create_parameters_for_grid_search(),
+                                      scoring='r2',
+                                      n_jobs=-1,
+                                      cv=4,
```

Comment:
Here and below for RF. We need to be careful with the cv param. In case we have a dataset with less than 4 labels for the target this would crash. I would simply add in the infobox when running the app locally something like: "The tuned versions require the targets to have at least 4 labels". You should also validate this in the backend and raise a custom exceeption if number_labels < 4